sync_diff_inspector/diff.go

```diff
@@ -537,6 +535,24 @@ func (df *Diff) GetMatchTable(db DBConfig, schema, table string, allTables map[s
 	return tableNames, nil
 }
 
+// RemoveMatchTable removes all the matched table.
+func (df *Diff) RemoveMatchTable(table string, allTables map[string]struct{}) map[string]struct{} {
+	tableNames := allTables
+
+	if table[0] == '~' {
+		tableRegex := regexp.MustCompile(fmt.Sprintf("(?i)%s", table[1:]))
```

Comment:
How can we ensure the regexp patten is always a valid expression

Reply:
BTW, do we need to ensure the expression should match the whole table name by alway add "^$" to the expression？

Reply:
> How can we ensure the regexp patten is always a valid expression
sync-diff will panic directly in the past. Now I have updated this part to report an error

Reply:
> BTW, do we need to ensure the expression should match the whole table name by alway add "^$" to the expression？

I think we should. But dm doesn't have this rule either. Maybe we could have a further discussion about this?